src/Persistence/Ui.php

```diff
@@ -19,35 +23,26 @@
  */
 class Ui extends \atk4\data\Persistence
 {
-    public $date_format = 'M d, Y';
 
-    public $time_format = 'H:i';
-
-    public $datetime_format = 'M d, Y H:i:s';
-    // 'D, d M Y H:i:s O';
-
-    /**
-     * Calendar input first day of week.
-     *  0 = sunday;.
-     *
-     * @var int
-     */
-    public $firstDayOfWeek = 0;
-
-    public $currency = '€';
-
-    /**
-     * Default decimal count for type 'money'
-     *  Used directly in number_format() second parameter.
-     *
-     * @var int
-     */
-    public $currency_decimals = 2;
-
-    public $yes = 'Yes';
-    public $no = 'No';
+    public $typeClass = [
+        'boolean' => Boolean::class,
+        'date' => Date::class,
+        'time' => Date::class,
+        'datetime' => Date::class,
+        'money' => Money::class,
+        'array' => Serial::class,
+        'object' => Serial::class,
+    ];
+
+    public function getTypeClass(string $type): ?string
+    {
+        return $this->typeClass[$type] ?? null;
+    }
 
-    public $calendar_options = [];
+    public function setTypeClass(string $type, string $class)
```

Comment:
registerTypeClass or addTypeClass

see how DBAL registry works (yes, they are for php <->, but usage should be about the same)

Reply:
done! I used registerTypeClass